streets_reader/src/split_ways.rs

```diff
@@ -24,19 +24,30 @@ pub fn split_up_roads(
 ) -> Output {
     timer.start("splitting up roads");
 
-    let mut roundabout_centers: HashMap<osm::NodeID, Pt2D> = HashMap::new();
-    let mut pt_to_intersection: HashMap<HashablePt2D, osm::NodeID> = HashMap::new();
+    let mut roundabout_centers: Vec<(Pt2D, Vec<osm::NodeID>)> = Vec::new();
+    // Note we iterate over this later and assign IDs based on the order, so HashMap would be
```

Comment:
I also rewrote a little logic here. I had a bug at first here using HashMap. Note with BTreeMap, the internal order of our `IntersectionIDs` will follow spatial patterns based on how `Pt2D` is ordered by X and then Y. That shouldn't matter, just kind of worth mentioning